bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieDecommissionTest.java

```diff
@@ -86,7 +91,18 @@ public void testDecommissionBookie() throws Exception {
          * this decommisionBookie should make sure that there are no
          * underreplicated ledgers because of this bookie
          */
-        bkAdmin.decommissionBookie(Bookie.getBookieAddress(killedBookieConf));
+        bkAdmin.decommissionBookie(
+            Bookie.getBookieAddress(killedBookieConf),
+            () -> {
+                Auditor auditor = recoveryProcess.getAuditor();
+                if (null != auditor) {
```

Comment:
why do we have this if block? in which case it can be null and is it ok to be null for our testscenario?

Reply:
this is just for safety guard

Reply:
hmm..in hypothetical scenario if it is null, then aren't we just ignoring it? is that ok to just ignore the fact that auditor is null and proceed with the test logic without waiting for auditor to do its auditing bookies task? 

Anyhow, I'm still not convinced how can we be ok with Auditor to be null? If there is no Auditor, then there is no replication process, right?

Reply:
> If there is no Auditor, then there is no replication process, right?

If there is no auditor, there is no replication process then the test will fail.

The logic here is just to avoid throw NPE (it is a coding style problem). I can change it to checkNotNull(auditor) to explicitly throw NPE if you like.

Reply:
since recoveryProcess.start() is async call here, as we discuss wait until auditor is not null here.